In [7]:
import pypsa
import pandas as pd

# Process and analysis

## Network without NEP

In [ ]:
n_no_nep = pypsa.Network("../results/transmission-derivation/baseline-nonep/prenetworks/base_s_52_lv1.11___2035.nc")

INFO:pypsa.io:Imported network base_s_52_lv1.11___2035.nc has buses, carriers, generators, global_constraints, lines, links, loads, storage_units, stores


In [448]:
ac_lines = n_no_nep.lines

In [449]:
line_cap = ac_lines.s_nom.mul(ac_lines.length).div(1e3).sum()
line_cap

133984.1929275

In [450]:
ac_lines.s_nom.sum()

577332.5445363616

In [451]:
dc_links = n_no_nep.links.query("carrier=='DC'")
dc_links.head()

bus0    bus1 type carrier  efficiency  active  \
Link                                                                       
relation/2505320-400-DC   DK1 0  DE0 10           DC    0.973880    True   
relation/3391794-200-DC   IT2 0   FR3 0           DC    0.971945    True   
relation/3392010-450-DC   PL0 0   SE1 0           DC    0.952022    True   
relation/3918230-450-DC  DE0 26   SE1 0           DC    0.956318    True   
relation/5487095-400-DC   DK0 0   DK1 0           DC    0.975062    True   

                         build_year  lifetime  p_nom  p_nom_mod  ...  \
Link                                                             ...   
relation/2505320-400-DC           0       inf  600.0        0.0  ...   
relation/3391794-200-DC           0       inf  300.0        0.0  ...   
relation/3392010-450-DC           0       inf  600.0        0.0  ...   
relation/3918230-450-DC           0       inf  600.0        0.0  ...   
relation/5487095-400-DC           0       inf  600.0        0.0  ...   

                         length_original  under_construction  voltage  \
Link                                                                    
relation/2505320-400-DC       269.210494                 0.0    400.0   
relation/3391794-200-DC       354.710531                 0.0    200.0   
relation/3392010-450-DC      1244.778527                 0.0    450.0   
relation/3918230-450-DC      1051.298387                 0.0    450.0   
relation/5487095-400-DC       217.084525                 0.0    400.0   

                         underground  \
Link                                   
relation/2505320-400-DC          1.0   
relation/3391794-200-DC          1.0   
relation/3392010-450-DC          1.0   
relation/3918230-450-DC          1.0   
relation/5487095-400-DC          1.0   

                                                                  geometry  \
Link                                                                         
relation/2505320-400-DC  LINESTRING (12.0074143643741 55.45030397887522...   
relation/3391794-200-DC  LINESTRING (8.713917011839937 40.6520491271003...   
relation/3392010-450-DC  LINESTRING (14.84028235523815 56.1530147407003...   
relation/3918230-450-DC  LINESTRING (10.80243095002194 53.8961375941516...   
relation/5487095-400-DC  LINESTRING (10.505724427906852 55.365970143543...   

                          dc  underwater_fraction  location              tags  \
Link                                                                            
relation/2505320-400-DC  1.0             0.281105            relation/2505320   
relation/3391794-200-DC  1.0             0.052118            relation/3391794   
relation/3392010-450-DC  1.0             0.940780            relation/3392010   
relation/3918230-450-DC  1.0             0.927793            relation/3918230   
relation/5487095-400-DC  1.0             0.560249            relation/5487095   

                         reversed  
Link                               
relation/2505320-400-DC     False  
relation/3391794-200-DC     False  
relation/3392010-450-DC     False  
relation/3918230-450-DC     False  
relation/5487095-400-DC     False  

[5 rows x 50 columns]

In [452]:
dc_links_cap = dc_links.p_nom.mul(dc_links.length).div(1e3).sum() # TWkm
dc_links_cap

3711.437760191092

In [453]:
grid_cap_wo_nep = line_cap + dc_links_cap
grid_cap_wo_nep

137695.6306876911

## Network with NEP lines

In [454]:
n_nep = pypsa.Network("../results/transmission-derivation/2032-confirmed-neplines/prenetworks/base_s_52_lv1.11___2035.nc")

INFO:pypsa.io:Imported network base_s_52_lv1.11___2035.nc has buses, carriers, generators, global_constraints, lines, links, loads, storage_units, stores


In [455]:
n_nep.lines

bus0    bus1                         type          x         r    g  \
Line                                                                         
0     AT0 0   CH0 0  Al/St 240/40 4-bundle 380.0  38.365333  5.668323  0.0   
1     AT0 0   CZ0 0  Al/St 240/40 4-bundle 380.0  22.331495  3.299389  0.0   
10    BE0 0   NL0 0  Al/St 240/40 4-bundle 380.0   9.227679  1.125321  0.0   
100   IT0 1   IT0 2  Al/St 240/40 4-bundle 380.0   6.556203  1.002476  0.0   
11    CH0 0  DE0 18  Al/St 240/40 4-bundle 380.0   2.432578  0.336994  0.0   
...     ...     ...                          ...        ...       ...  ...   
95    FR0 2   FR0 4  Al/St 240/40 4-bundle 380.0   6.364712  1.006031  0.0   
96    FR0 2   IT0 1  Al/St 240/40 4-bundle 380.0  24.252374  3.225714  0.0   
97    FR0 3   FR0 4  Al/St 240/40 4-bundle 380.0  13.282543  1.890312  0.0   
98    FR0 4   FR0 5  Al/St 240/40 4-bundle 380.0   9.695653  1.523438  0.0   
99    IT0 0   IT0 2  Al/St 240/40 4-bundle 380.0  23.756093  3.038496  0.0   

             b         s_nom  s_nom_mod  s_nom_extendable  ...  x_pu  r_pu  \
Line                                                       ...               
0     0.009368   4379.317262        0.0              True  ...   0.0   0.0   
1     0.005452   4379.317262        0.0              True  ...   0.0   0.0   
10    0.005855  10188.615670        0.0              True  ...   0.0   0.0   
100   0.019190  14121.063824        0.0              True  ...   0.0   0.0   
11    0.011205  21628.464844        0.0              True  ...   0.0   0.0   
...        ...           ...        ...               ...  ...   ...   ...   
95    0.011275  10668.999962        0.0              True  ...   0.0   0.0   
96    0.006373   5865.157047        0.0              True  ...   0.0   0.0   
97    0.029920  15025.973768        0.0              True  ...   0.0   0.0   
98    0.009117   7876.068035        0.0              True  ...   0.0   0.0   
99    0.019710  10680.171690        0.0              True  ...   0.0   0.0   

      g_pu  b_pu  x_pu_eff  r_pu_eff  s_nom_opt   dc i_nom  v_nom  
Line                                                               
0      0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
1      0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
10     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
100    0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
11     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
...    ...   ...       ...       ...        ...  ...   ...    ...  
95     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
96     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
97     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
98     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
99     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  

[101 rows x 34 columns]

In [456]:
ac_lines = n_nep.lines

In [457]:
line_cap = ac_lines.s_nom.mul(ac_lines.length).div(1e3).sum()
line_cap

135833.54580619605

In [458]:
dc_links = n_nep.links.query("carrier=='DC'")
dc_links.head()

bus0   bus1 type carrier  efficiency  active  \
Link                                                                      
relation/2505320-400-DC   DK1 0  DE0 7           DC    0.974807    True   
relation/3391794-200-DC   IT2 0  FR3 0           DC    0.971945    True   
relation/3392010-450-DC   PL0 0  SE1 0           DC    0.952022    True   
relation/3918230-450-DC  DE0 13  SE1 0           DC    0.956412    True   
relation/5487095-400-DC   DK0 0  DK1 0           DC    0.975062    True   

                         build_year  lifetime  p_nom  p_nom_mod  ...  \
Link                                                             ...   
relation/2505320-400-DC           0       inf  600.0        0.0  ...   
relation/3391794-200-DC           0       inf  300.0        0.0  ...   
relation/3392010-450-DC           0       inf  600.0        0.0  ...   
relation/3918230-450-DC           0       inf  600.0        0.0  ...   
relation/5487095-400-DC           0       inf  600.0        0.0  ...   

                         under_construction  voltage  project_status  \
Link                                                                   
relation/2505320-400-DC                 0.0    400.0                   
relation/3391794-200-DC                 0.0    200.0                   
relation/3392010-450-DC                 0.0    450.0                   
relation/3918230-450-DC                 0.0    450.0                   
relation/5487095-400-DC                 0.0    400.0                   

                                     tags  underground  \
Link                                                     
relation/2505320-400-DC  relation/2505320          1.0   
relation/3391794-200-DC  relation/3391794          1.0   
relation/3392010-450-DC  relation/3392010          1.0   
relation/3918230-450-DC  relation/3918230          1.0   
relation/5487095-400-DC  relation/5487095          1.0   

                                                                  geometry  \
Link                                                                         
relation/2505320-400-DC  LINESTRING (12.0074143643741 55.45030397887522...   
relation/3391794-200-DC  LINESTRING (8.713917011839937 40.6520491271003...   
relation/3392010-450-DC  LINESTRING (14.84028235523815 56.1530147407003...   
relation/3918230-450-DC  LINESTRING (10.80243095002194 53.8961375941516...   
relation/5487095-400-DC  LINESTRING (10.505724427906852 55.365970143543...   

                          dc  location  underwater_fraction  reversed  
Link                                                                   
relation/2505320-400-DC  1.0                       0.281105     False  
relation/3391794-200-DC  1.0                       0.052118     False  
relation/3392010-450-DC  1.0                       0.940780     False  
relation/3918230-450-DC  1.0                       0.927793     False  
relation/5487095-400-DC  1.0                       0.560249     False  

[5 rows x 51 columns]

In [459]:
dc_links_cap = dc_links.p_nom.mul(dc_links.length).div(1e3).sum() # TWkm
dc_links_cap

3684.3793361037806

In [460]:
grid_cap_w_nep_lines = line_cap + dc_links_cap
grid_cap_w_nep_lines

139517.92514229982

## Network with NEP lines + links

In [461]:
n_nep_lines_links = pypsa.Network("../results/transmission-derivation/2032-confirmed-neplinesandlinks/prenetworks/base_s_52_lv1.11___2035.nc")

INFO:pypsa.io:Imported network base_s_52_lv1.11___2035.nc has buses, carriers, generators, global_constraints, lines, links, loads, storage_units, stores


In [462]:
n_nep_lines_links.lines

bus0    bus1                         type          x         r    g  \
Line                                                                         
0     AT0 0   CH0 0  Al/St 240/40 4-bundle 380.0  38.365333  5.668323  0.0   
1     AT0 0   CZ0 0  Al/St 240/40 4-bundle 380.0  22.331495  3.299389  0.0   
10    BE0 0   NL0 0  Al/St 240/40 4-bundle 380.0   9.227679  1.125321  0.0   
100   IT0 1   IT0 2  Al/St 240/40 4-bundle 380.0   6.556203  1.002476  0.0   
11    CH0 0  DE0 18  Al/St 240/40 4-bundle 380.0   2.432578  0.336994  0.0   
...     ...     ...                          ...        ...       ...  ...   
95    FR0 2   FR0 4  Al/St 240/40 4-bundle 380.0   6.364712  1.006031  0.0   
96    FR0 2   IT0 1  Al/St 240/40 4-bundle 380.0  24.252374  3.225714  0.0   
97    FR0 3   FR0 4  Al/St 240/40 4-bundle 380.0  13.282543  1.890312  0.0   
98    FR0 4   FR0 5  Al/St 240/40 4-bundle 380.0   9.695653  1.523438  0.0   
99    IT0 0   IT0 2  Al/St 240/40 4-bundle 380.0  23.756093  3.038496  0.0   

             b         s_nom  s_nom_mod  s_nom_extendable  ...  x_pu  r_pu  \
Line                                                       ...               
0     0.009368   4379.317262        0.0              True  ...   0.0   0.0   
1     0.005452   4379.317262        0.0              True  ...   0.0   0.0   
10    0.005855  10188.615670        0.0              True  ...   0.0   0.0   
100   0.019190  14121.063824        0.0              True  ...   0.0   0.0   
11    0.011205  21628.464844        0.0              True  ...   0.0   0.0   
...        ...           ...        ...               ...  ...   ...   ...   
95    0.011275  10668.999962        0.0              True  ...   0.0   0.0   
96    0.006373   5865.157047        0.0              True  ...   0.0   0.0   
97    0.029920  15025.973768        0.0              True  ...   0.0   0.0   
98    0.009117   7876.068035        0.0              True  ...   0.0   0.0   
99    0.019710  10680.171690        0.0              True  ...   0.0   0.0   

      g_pu  b_pu  x_pu_eff  r_pu_eff  s_nom_opt   dc i_nom  v_nom  
Line                                                               
0      0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
1      0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
10     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
100    0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
11     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
...    ...   ...       ...       ...        ...  ...   ...    ...  
95     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
96     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
97     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
98     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  
99     0.0   0.0       0.0       0.0        0.0  0.0  2.58    380  

[101 rows x 34 columns]

In [463]:
ac_lines = n_nep_lines_links.lines

In [464]:
line_cap = ac_lines.s_nom.mul(ac_lines.length).div(1e3).sum()
line_cap

135833.54580619605

In [465]:
dc_links = n_nep_lines_links.links.query("carrier=='DC'")
dc_links.head()

bus0   bus1 type carrier  efficiency  active  \
Link                                                                      
relation/2505320-400-DC   DK1 0  DE0 7           DC    0.974807    True   
relation/3391794-200-DC   IT2 0  FR3 0           DC    0.971945    True   
relation/3392010-450-DC   PL0 0  SE1 0           DC    0.952022    True   
relation/3918230-450-DC  DE0 13  SE1 0           DC    0.956412    True   
relation/5487095-400-DC   DK0 0  DK1 0           DC    0.975062    True   

                         build_year  lifetime  p_nom  p_nom_mod  ...  \
Link                                                             ...   
relation/2505320-400-DC           0       inf  600.0        0.0  ...   
relation/3391794-200-DC           0       inf  300.0        0.0  ...   
relation/3392010-450-DC           0       inf  600.0        0.0  ...   
relation/3918230-450-DC           0       inf  600.0        0.0  ...   
relation/5487095-400-DC           0       inf  600.0        0.0  ...   

                         under_construction  voltage  project_status  \
Link                                                                   
relation/2505320-400-DC                 0.0    400.0                   
relation/3391794-200-DC                 0.0    200.0                   
relation/3392010-450-DC                 0.0    450.0                   
relation/3918230-450-DC                 0.0    450.0                   
relation/5487095-400-DC                 0.0    400.0                   

                                     tags  underground  \
Link                                                     
relation/2505320-400-DC  relation/2505320          1.0   
relation/3391794-200-DC  relation/3391794          1.0   
relation/3392010-450-DC  relation/3392010          1.0   
relation/3918230-450-DC  relation/3918230          1.0   
relation/5487095-400-DC  relation/5487095          1.0   

                                                                  geometry  \
Link                                                                         
relation/2505320-400-DC  LINESTRING (12.0074143643741 55.45030397887522...   
relation/3391794-200-DC  LINESTRING (8.713917011839937 40.6520491271003...   
relation/3392010-450-DC  LINESTRING (14.84028235523815 56.1530147407003...   
relation/3918230-450-DC  LINESTRING (10.80243095002194 53.8961375941516...   
relation/5487095-400-DC  LINESTRING (10.505724427906852 55.365970143543...   

                          dc  location  underwater_fraction  reversed  
Link                                                                   
relation/2505320-400-DC  1.0                       0.281105     False  
relation/3391794-200-DC  1.0                       0.052118     False  
relation/3392010-450-DC  1.0                       0.940780     False  
relation/3918230-450-DC  1.0                       0.927793     False  
relation/5487095-400-DC  1.0                       0.560249     False  

[5 rows x 51 columns]

In [466]:
dc_links_cap = dc_links.p_nom.mul(dc_links.length).div(1e3).sum() # TWkm
dc_links_cap

5447.931700137143

In [467]:
grid_cap_w_nep_lines_links = line_cap + dc_links_cap
grid_cap_w_nep_lines_links

141281.4775063332

## Calculate ratios

### Only confirmed

In [354]:
# Ratio with NEP lines + links and without NEP
grid_cap_w_nep_lines_links/grid_cap_wo_nep

1.1158646946438133

In [356]:
# Ratio with NEP lines and without NEP
grid_cap_w_nep_lines/grid_cap_wo_nep

1.0461220566935823

In [355]:
# Ratio with NEP lines + links and NEP lines
grid_cap_w_nep_lines_links/grid_cap_w_nep_lines

1.0666677826970425

### Only confirmed + 2032

In [468]:
# Ratio with NEP lines + links and without NEP
grid_cap_w_nep_lines_links/grid_cap_wo_nep

1.0260418344484379

In [469]:
# Ratio with NEP lines and without NEP
grid_cap_w_nep_lines/grid_cap_wo_nep

1.0132342213438994

In [470]:
# Ratio with NEP lines + links and NEP lines
grid_cap_w_nep_lines_links/grid_cap_w_nep_lines

1.0126403282032375

### NEP baseline

In [337]:
# Ratio with NEP lines + links and without NEP
grid_cap_w_nep_lines_links/grid_cap_wo_nep

1.163350733136277

In [339]:
# Ratio with NEP lines and without NEP
grid_cap_w_nep_lines/grid_cap_wo_nep

1.0461220566935823

In [338]:
# Ratio with NEP lines + links and NEP lines
grid_cap_w_nep_lines_links/grid_cap_w_nep_lines

1.112060228242594

## Carbon emission targets

In [57]:
ctrys = ['AT', 'BE', 'CH', 'CZ', 'DE', 'DK', 'FR', 'IT', 'LU', 'NL', 'PL', 'SE']

In [58]:

# 1990 emissions in MtCO2
co2_totals = pd.read_csv("../data/co2_totals_1990.csv", index_col=0).loc[ctrys]

In [59]:
# 2005 emissions in MtCO2
co2_totals_2005 = pd.read_csv("../data/co2_totals_2005.csv", index_col=0).loc[ctrys]

In [60]:
# Total emissions excl. LULUCF
co2_totals_excl_lulucf = co2_totals.drop(["LULUCF"], axis=1).sum(axis=1)
co2_totals_excl_lulucf

Country_code
AT      62.124701
BE     120.309343
CH      44.534441
CZ     166.068823
DE    1052.348544
DK      54.685986
FR     400.777042
IT     438.008802
LU      11.847635
NL     163.302301
PL     376.546518
SE      57.348825
dtype: float64

In [61]:
# Total emissions incl. LULUCF
co2_totals_incl_lulucf = co2_totals.sum(axis=1)
co2_totals_incl_lulucf

Country_code
AT      49.968104
BE     117.060288
CH      42.515838
CZ     160.298010
DE    1021.884333
DK      60.858777
FR     374.572263
IT     432.346810
LU      11.927560
NL     169.786966
PL     343.554029
SE      21.112932
dtype: float64

In [62]:
# Total emissions excl. LULUCF
co2_totals_excl_lulucf_2005 = co2_totals_2005.drop(["LULUCF"], axis=1).sum(axis=1)
co2_totals_excl_lulucf_2005

Country_code
AT     79.192432
BE    125.661688
CH     45.915746
CZ    126.738399
DE    866.388549
DK     52.164863
FR    426.412463
IT    500.005676
LU     12.105448
NL    177.897237
PL    323.161267
SE     53.852857
dtype: float64

In [63]:
# Total emissions incl. LULUCF 2005
co2_totals_incl_lulucf_2005 = co2_totals_2005.sum(axis=1)
co2_totals_incl_lulucf_2005

Country_code
AT     68.414161
BE    124.024742
CH     43.712103
CZ    119.316791
DE    851.168952
DK     57.805099
FR    377.355895
IT    463.962937
LU     11.462305
NL    183.428661
PL    271.748435
SE     19.936933
dtype: float64

In [64]:
co2_totals_2005_all = pd.read_csv("../data/co2_totals_2005.csv", index_col=0)
co2_totals_excl_lulucf_2005_all = co2_totals_2005_all.drop(["LULUCF"], axis=1).sum(axis=1)

In [65]:
co2_totals_2005_all = pd.read_csv("../data/co2_totals_2005.csv", index_col=0)
co2_totals_incl_lulucf_2005_all = co2_totals_2005_all.sum(axis=1)

In [66]:
emission_share_included_ctrys_excl_lulucf = co2_totals_excl_lulucf_2005.sum()/co2_totals_excl_lulucf_2005_all.sum()
emission_share_included_ctrys_excl_lulucf

0.6245472968134671

In [67]:
emission_share_included_ctrys_incl_lulucf = co2_totals_incl_lulucf_2005.sum()/co2_totals_incl_lulucf_2005_all.sum()
emission_share_included_ctrys_incl_lulucf

0.6327636937085288

### ESR targets 2030 (compared to 2005) [Source](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en)

In [68]:
esr_targets = {
    "AT": 0.48,
    "BE": 0.47,
    "CZ": 0.26,
    "DE": 0.5,
    "DK": 0.5,
    "FR": 0.475,
    "IT": 0.437,
    "LU": 0.5,
    "NL": 0.48,
    "PL": 0.177,
    "SE": 0.5,
}

### Country specific emission reduction targets

| Country | 2030 | 2035 | 2040 | 2045 | 2050 | Source |
|---------|------|------|------|------|------|--------|
| DE | 65% below 1990 excl. LULUCF | - | 85% below 1990 excl. LULUCF | 100% | - | [Source](https://www.bundesregierung.de/breg-de/themen/klimaschutz/klimaschutzgesetz-2021-1913672) |
| AT | 48% below 2005 excl. LULUCF | - | 100% | - | - | [Source1](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) [Source2](https://www.sustainability.gov/pdfs/austria-nzgi-roadmap.pdf) |
| BE | 55% below 1990 excl. LULUCF | - | - | - | 100% | [Source](https://www.sustainability.gov/pdfs/belgium-nzgi-roadmap.pdf) |
| CH | 50% below 1990 excl. LULUCF | - | - |  - | 100% | [Source](https://www.bafu.admin.ch/bafu/en/home/topics/climate/info-specialists/emission-reduction/reduction-targets.html) |
| CZ | 26% below 2005 excl. LULUCF | - | - |  - | 100% | [Source](https://tracker.carbongap.org/region/czech-republic/) |
| DK | 70% below 1990 incl. LULUCF and 50% below 2005 excl. LULUCF | - | - |  100% | 110% | [Source](https://unfccc.int/sites/default/files/resource/SBI60_MA_DNK-7June2024-Presentation.pdf) |
| FR | 47.5% below 2005 excl. LULUCF | - | - |  - | 100% | [Source](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) |
| IT | 43.7% below 2005 excl. LULUCF | - | - |  - | 100% | [Source1](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) [Source2](https://www.iea.org/countries/italy#) |
| LU | 55% below 2005 excl. LULUCF | - | - |  - | 100% | [Source](https://gouvernement.lu/en/dossiers/2023/2023-pnec.html) |
| NL | 55% below 1990 | - | - |  - | 100% | [Source](https://commission.europa.eu/document/download/b6d21e56-4297-4b91-a692-300716209f72_en?filename=NL_FINAL%20UPDATED%20NECP%202021-2030%20%28English%29.pdf) |
| PL | 17.7% below 2005 excl. LULUCF | - | - |  - |  | [Source](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en) |
| SE | 63% below 1990 | - | 75% below 1990 |  100% |  | [Source](https://www.bafu.admin.ch/bafu/en/home/topics/climate/info-specialists/emission-reduction/reduction-targets.html) |

In [69]:
# calculate linearly interpolated 2035 targets
# base 1990
target2035_base1990 = dict(DE = 0.75,
BE = 0.55 + (1-0.55)*(2035-2030)/(2050-2030),
CH = 0.625,
DK = 0.7 + (1-0.7)*(2035-2030)/(2045-2030),
NL = 0.55 + (1-0.55)*(2035-2030)/(2050-2030),
SE = 0.69,
PL = (esr_targets["PL"]/esr_targets["DE"])*0.75
)
# base 2005
# Polands 2035 target is deduced based on the ESR distrubution key in relation to the German interpolated target
target2035_base2005 = dict(AT = 0.74,
CZ = 0.26 + (1-0.26)*(2035-2030)/(2050-2030),
FR = 0.475 + (1-0.475)*(2035-2030)/(2050-2030),
IT = 0.437 + (1-0.437)*(2035-2030)/(2050-2030),
LU = 0.55 + (1-0.55)*(2035-2030)/(2050-2030)
)
print(target2035_base1990, target2035_base2005)

{'DE': 0.75, 'BE': 0.6625000000000001, 'CH': 0.625, 'DK': 0.7999999999999999, 'NL': 0.6625000000000001, 'SE': 0.69, 'PL': 0.26549999999999996} {'AT': 0.74, 'CZ': 0.445, 'FR': 0.60625, 'IT': 0.57775, 'LU': 0.6625000000000001}


In [70]:
target_2035 = pd.DataFrame(index=co2_totals.index)
target_2035["base1990"] = target2035_base1990
target_2035["base2005"] = target2035_base2005
target_2035

base1990  base2005
Country_code                    
AT                 NaN   0.74000
BE              0.6625       NaN
CH              0.6250       NaN
CZ                 NaN   0.44500
DE              0.7500       NaN
DK              0.8000       NaN
FR                 NaN   0.60625
IT                 NaN   0.57775
LU                 NaN   0.66250
NL              0.6625       NaN
PL              0.2655       NaN
SE              0.6900       NaN

In [71]:
# Calculate aggregated 1990 emissions in MtCO2
co2_1990_aggregated = co2_totals_excl_lulucf.sum()
co2_1990_aggregated

2947.9029606100003

In [72]:
# Calculate absolute target emissions in 2035 based on shares
target_2035["absolute_emissions"] = ((1-target_2035.base1990) * co2_totals_excl_lulucf).combine_first((1-target_2035.base2005) * co2_totals_excl_lulucf_2005)
target_2035_aggregated = target_2035.absolute_emissions.sum()
target_2035_aggregated

1154.8379684469976

In [73]:
# Calculate share of 1990 emissions
round(target_2035_aggregated/co2_1990_aggregated,3)

0.392

Alternate approach: For missing values take interpolated emission reduction target for Germany 2035 (75%). For other countries take distribution key from [EU effort sharing regulation](https://climate.ec.europa.eu/eu-action/effort-sharing-member-states-emission-targets/effort-sharing-2021-2030-targets-and-flexibilities_en). Then calculate total emissions based on 1990 values and weighted emission reduction target.

In [74]:
target_2035

base1990  base2005  absolute_emissions
Country_code                                        
AT                 NaN   0.74000           20.590032
BE              0.6625       NaN           40.604403
CH              0.6250       NaN           16.700415
CZ                 NaN   0.44500           70.339812
DE              0.7500       NaN          263.087136
DK              0.8000       NaN           10.937197
FR                 NaN   0.60625          167.899907
IT                 NaN   0.57775          211.127397
LU                 NaN   0.66250            4.085589
NL              0.6625       NaN           55.114527
PL              0.2655       NaN          276.573417
SE              0.6900       NaN           17.778136

## Europe's 2040 climate target
Source: https://eur-lex.europa.eu/resource.html?uri=cellar:6c154426-c5a6-11ee-95d9-01aa75ed71a1.0001.02/DOC_3&format=PDF

In [75]:
# calculate linearly interpolated 2035 targets
target_sector = pd.read_csv("../data/co2_total_sector.csv", index_col=0)
target_sector['2035 S2'] = (target_sector['2030'] + target_sector['2040 S2'])/2
target_sector_share = target_sector/target_sector.loc['Total Gross GHG Emissions',:]

In [76]:
target_sector

2005  2015  2030  2040 S1  2040 S2  2040 S3  \
MtC02-eq                                                                   
Total Gross GHG Emissions    4641  3914  2301     1273      943      748   
Power and district heating   1300  1012   339      123       42       23   
Other Energy sectors*         277   237   133       71       59       53   
Industry (Energy)             469   360   232      126       94       75   
Domestic Transport            822   772   583      190      143      120   
Residential and Services**    648   514   221      119       92       75   
Industry (Non-Energy)         343   233   157      139       88       14   
Other Non-Energv sectors***   101   130    56       33       26       25   
Intra-EU aviation              35    38    43       31       29       28   
Intra-EU navigation            31    27    25        7        6        4   
50% extra-EU maritime MRV      50    42    44       14       11        9   
Agriculture****               390   385   361      351      302      271   
Waste                         155   118    87       68       55       55   
C02 calibration                15    43    24        3       -1       -1   
Non-C02 calibration             5     2    -3       -3       -3       -3   

                             2040 LIFE  2050 S1  2050 S2  2050 S3  2050 LIFE  \
MtC02-eq                                                                       
Total Gross GHG Emissions          740      416      413      411     360.00   
Power and district heating          34       21       22       19      15.00   
Other Energy sectors*               57       39       39       38      36.00   
Industry (Energy)                   86        6        6        9      11.00   
Domestic Transport                 134       10        8        7       9.00   
Residential and Services**          92       20       19       19      29.00   
Industry (Non-Energy)               13        7        7        7       7.00   
Other Non-Energv sectors***         25       23       22       22      22.00   
Intra-EU aviation                   14       14       12       11      10.00   
Intra-EU navigation                  0        0        0        0       0.00   
50% extra-EU maritime MRV            0        0        0        0       0.00   
Agriculture****                    209      249      249      249     194.00   
Waste                               55       32       32       32       0.32   
C02 calibration                     -1        0        0        0       0.00   
Non-C02 calibration                 -3       -3       -3       -3      -3.00   

                             2035 S2  
MtC02-eq                              
Total Gross GHG Emissions     1622.0  
Power and district heating     190.5  
Other Energy sectors*           96.0  
Industry (Energy)              163.0  
Domestic Transport             363.0  
Residential and Services**     156.5  
Industry (Non-Energy)          122.5  
Other Non-Energv sectors***     41.0  
Intra-EU aviation               36.0  
Intra-EU navigation             15.5  
50% extra-EU maritime MRV       27.5  
Agriculture****                331.5  
Waste                           71.0  
C02 calibration                 11.5  
Non-C02 calibration             -3.0

In [80]:
# Select the included sector in the model (other energy sectors include non-CO2 GHG emissions and are thus excluded)
include_sector = ['Power and district heating','Residential and Services**','Other Energy sectors*']

target_sector_share_2035 = target_sector_share.loc[include_sector,'2035 S2'].sum()
target_sector_share_2035

0.27311960542540076

In [81]:
target_2035['included_sector_emissions'] = target_2035['absolute_emissions'] * target_sector_share_2035
target_2035

base1990  base2005  absolute_emissions  \
Country_code                                           
AT                 NaN   0.74000           20.590032   
BE              0.6625       NaN           40.604403   
CH              0.6250       NaN           16.700415   
CZ                 NaN   0.44500           70.339812   
DE              0.7500       NaN          263.087136   
DK              0.8000       NaN           10.937197   
FR                 NaN   0.60625          167.899907   
IT                 NaN   0.57775          211.127397   
LU                 NaN   0.66250            4.085589   
NL              0.6625       NaN           55.114527   
PL              0.2655       NaN          276.573417   
SE              0.6900       NaN           17.778136   

              included_sector_emissions  
Country_code                             
AT                             5.623541  
BE                            11.089859  
CH                             4.561211  
CZ                            19.211182  
DE                            71.854255  
DK                             2.987163  
FR                            45.856756  
IT                            57.663031  
LU                             1.115854  
NL                            15.052858  
PL                            75.537623  
SE                             4.855557

In [85]:
# Calculate share of 1990 emissions
round(target_2035['included_sector_emissions'].sum()/co2_1990_aggregated,3)

0.107

# Convert cost assumptions to NPVs

In [86]:
import numpy as np

In [87]:
def convert_to_NPV(value = 23_500, i = 0.02, year = 2030, reference_year = 2020):
    NPV = value/((1+i)**(year-reference_year))
    return NPV

In [88]:
convert_to_NPV(value = 23_500, i = 0.02, year = 2024, reference_year = 2020)

21710.367511623084

In [89]:
convert_to_NPV(value = 20_000, i = 0.02, year = 2024, reference_year = 2020)

18476.908520530284

In [120]:
convert_to_NPV(value = 15_250, i = 0.02, year = 2024, reference_year = 2020)

14088.642746904343

In [91]:
convert_to_NPV(value = 30_000, i = 0.02, year = 2022, reference_year = 2020)

28835.06343713956

In [92]:
convert_to_NPV(value = 1_725_000, i = 0.02, year = 2022, reference_year = 2020)

1658016.1476355249